### PR TITLE
Manager multichain and workflow config builder

### PR DESCRIPTION
**Pull request by Art3miX**
Added a builder to workflow config, so now its nicer to build the workflow config before init.
Change the `setup_manager` in local-ci to be multichain, so now when `setup_manager` is called, you need to provide it the chain config file (in configs folder) and the contracts name (package name, the name as in the artifacts folder), basically the services this test needs, like discussed before.

Regarding contract names, we have a special care for `IBC_TRANSFER` mainly because we need to upload a specific contract on neutron, and generic one on other chains, this is taken care in `get_contract_path` basically manually.
so if you want to provide `ibc_transfer` to `setup_manager` you can choose one of the 3 names:
1. valence-neutron-ibc-transfer-service
2. valence-generic-ibc-transfer-service
3. ibc_transfer
 
`get_contract_path` will upload the correct contract on the correct chain regardless of the name provided.

In production we will manually upload each contract to the correct chain, and provide the code id in config file.

P.S something nice about the builder, is if you added an account or service, and didn't use it, it will give you a warning of unused_variables.

### CODE REVIEW

File: packages/service-utils/src/lib.rs [hunk 1]
```diff
@@ -58,7 +58,51 @@ impl From<&str> for ServiceAccountType {
...
+    // pub fn get_id(&self) -> StdResult<Id> {
+    //     match self {
+    //         ServiceAccountType::Addr(_) => {
+    //             Err(StdError::generic_err("ServiceAccountType is an address"))
+    //         }
+    //         ServiceAccountType::AccountId(id) => Ok(*id),
+    //         ServiceAccountType::ServiceId(id) => Ok(*id),
+    //     }
+    // }
+
```
Comment (keyleu) [inline packages/service-utils/src/lib.rs:105]: prob leftover?

File: local-interchaintest/src/utils/manager.rs [hunk 2]
```diff
@@ -151,70 +134,158 @@ pub fn setup_manager(test_ctx: &mut TestContext) -> Result<(), Box<dyn Error>> {
     // let code_ids = GLOBAL_CONFIG.read().unwrap().contracts.code_ids.clone();
     // println!("{:?}", code_ids);
```
Comment (keyleu) [inline local-interchaintest/src/utils/manager.rs:135]: can prob delete this part

File: local-interchaintest/src/utils/manager.rs [hunk 3]
```diff
@@ -13,133 +13,116 @@ use valence_workflow_manager::{
...
+pub const NEUTRON_IBC_TRANSFER_NAME: &str = "valence-neutron-ibc-transfer-service";
+pub const ASTROPORT_LPER_NAME: &str = "valence_astroport_lper";
+pub const ASTROPORT_WITHDRAWER_NAME: &str = "valence_astroport_withdrawer";
+
+/// Those contracts will always be uploaded because each workflow needs them
+const BASIC_CONTRACTS: [&str; 2] = [PROCESSOR_NAME, BASE_ACCOUNT_NAME];
+
+/// Setup everything that is needed for the manager to run, including uploading of the contracts
+///
+/// You can pass a list of contracts to upload, authorizaion, processor and base account are always updated,
```
Comment (keyleu) [inline local-interchaintest/src/utils/manager.rs:33]: typo: authorization

File: local-interchaintest/src/utils/manager.rs [hunk 4]
```diff
@@ -13,133 +13,116 @@ use valence_workflow_manager::{
...
     uploader.with_chain_name(NEUTRON_CHAIN_NAME);
 
+    // combine the basic contracts with the contracts passed
+    BASIC_CONTRACTS.iter().for_each(|s| {
+        if !contracts.contains(s) {
+            contracts.push(s);
+        }
+    });
+
+    // TODO: use a function to get the path of the correct contract based on the chain we are talking about
```
Comment (keyleu) [inline local-interchaintest/src/utils/manager.rs:57]: we're doing this already right?

**Review by keyleu — APPROVED** [review]
LGTM! Only left things regarding comments. Will wait for this to be merged before splitting the config files

**Comment by Art3miX** [comment]
Thanks @keyleu , fixed.

**Pull request merged**